subsys/bluetooth/host/att.c

```diff
@@ -378,11 +378,12 @@ static u8_t att_handle_rsp(struct bt_att *att, void *pdu, u16_t len, u8_t err)
 	att_req_destroy(att->req);
 	att->req = NULL;
 
-	func(att->chan.chan.conn, err, pdu, len, params);
-
 process:
 	/* Process pending requests */
 	att_process(att);
```

Comment:
I wonder why you didn't just add a check to att->req on att_process then?

Reply:
@Vudentz that would cause a new request to skip ahead in the request queue.

Reply:
Right, then lets have this one merged.